src/manager.S

```diff
@@ -0,0 +1,165 @@
+.data
+
+comps_cnt:     .dword 0
+comps_addr:    .dword 0
+switcher_caps: .dword 0
+
+.text
+.balign 4
+
+#include "comps_offsets.h"
+
+// Variables
+.global comps_cnt
+.global comps_addr
+.global switcher_caps
+
+// Functions
+.global asm_call_wrapper
+.global init_compartments
+.global add_compartment
+.global del_compartment
+
+/**
+ * Wrapper to store/restore state when coming from C
```

Comment:
I'm having a hard time seeing on the github interface what you are referring to, so I'll assume it's `asm_call_wrapper`. This function (and all functions in `manager.S`) are meant to only be called from a privileged environment, and I initially designed it to not have to save the `clr` in all internal functions, and to setup the registers properly for the eventual function call (in `x0`) it was supposed to call. I abandoned developing this, and, as the number of privileged functions are limited, it might be easier to just drop this? At the moment it is unused, so maybe just for that it should be deleted from this PR.

Reply:
If it's not used, then it's probably best to drop it, if only to simplify the analysis. (I hadn't spotted that it's not used.)

Reply:
The function `asm_call_wrapper` is now deleted in the latest version. Guess it's kinda hard to see with this Github UI.